modules/cli/src/main/java/org/apache/ignite/internal/cli/call/node/version/NodeVersion.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.cli.call.node.version;
+
+/**
+ * Represents the node version.
+ */
+public class NodeVersion {
+    private final String version;
+
+    private final String product;
+
+    /** Constructor. */
+    NodeVersion(String version, String product) {
```

Comment:
Both product and version are Strings. It is easy to mismatch the order and create `new Version("Ignite", "1.1.1")`. Can you add the validation or builder?